### PR TITLE
🐛 fix conversion-gen

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

Fixes https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/826#discussion_r606081071

- [x] squashed commits
- if necessary:
  - [x] includes documentation
  - [x] adds unit tests

/hold

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,7 @@ include $(ROOT_DIR_RELATIVE)/common.mk
 
 # Active module mode, as we use go modules to manage dependencies
 export GO111MODULE=on
+unexport GOPATH
```
Comment (sbueringer) [inline Makefile:24]: I though it doesn't make sense to support various variants of GOPATH setups when we can just "disable" GOPATH. As far as I can see it works.

**Comment by sbueringer** [comment]
@jichenjc please take a look, thx :)

**Comment by jichenjc** [comment]
> 
> 
> @jichenjc please take a look, thx :)

will check soon :) , thanks for the reminder

**Pull request merged**